packages/arb-avm-cpp/data_storage/src/value/value.cpp

```diff
@@ -43,6 +43,15 @@ struct ValueBeingParsed {
     }
 };
 
+constexpr uint64_t tupleInlineNumerator = 7;
+constexpr uint64_t tupleInlineDenominator = 8;
+bool shouldInlineTuple(const Tuple& tuple,
+                       const std::vector<unsigned char>& secret_hash_seed) {
+    auto hash = tuple.getHashPreImage().secretHash(secret_hash_seed);
+    return tuple.tuple_size() == 0 ||
+           (hash % tupleInlineDenominator) < tupleInlineNumerator;
```

Comment:
I wondering whether it would be better to have small tuples be more likely to inline than larger tuples. Likely not a change to make here, but an interesting possible future extension

Reply:
I'm not sure. A lot of trees in ArbOS are two-tuples right? It would definitely be an interesting experiment.

Reply:
I'm not totally sure. It'd definitely be interesting to do some analysis on the latest machine state on rinkeby or mainnet to get a better idea of the structure of the register